secrets/store.go

```diff
@@ -26,6 +26,7 @@ func nopSecretHandlerFunc(sec *Secrets) {}
 // memory so there's little performance impact to doing so and you will be sure
 // to always have the current version in the face of key rotation etc.
 type Store struct {
+	io.Closer
```

Comment:
style nit: Please add a blank line after this line (always add blank lines after embedded structs/interfaces).

But then there's no reason to embed `io.Closer` to `Store` struct. As long as you implemented that interface it's satisfied, that's not controlled by embedding.